repos/system_upgrade/common/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py

```diff
@@ -19,6 +17,30 @@
 HANA_BASE_PATH = '/hana/shared'
 HANA_SAPCONTROL_PATH = 'exe/linuxx86_64/hdb/sapcontrol'
 
+# map of expected upgrade paths per source system and flavour, expected
+# does not mean supported. Supported paths are checked later.
+upgrade_paths_map = {
+
+    # expected upgrade paths for RHEL 7
+    ('7.6', LEAPP_UPGRADE_FLAVOUR_DEFAULT): '8.4',
+    ('7.9', LEAPP_UPGRADE_FLAVOUR_DEFAULT): '8.4',
+    ('7.7', LEAPP_UPGRADE_FLAVOUR_SAP_HANA): '8.2',
+
+    # expected upgrade paths for RHEL 8
+    ('8.6', LEAPP_UPGRADE_FLAVOUR_DEFAULT): '9.0',
+
+    # unsupported fallback paths for RHEL 7
+    ('7', LEAPP_UPGRADE_FLAVOUR_DEFAULT): '8.4',
+    ('7', LEAPP_UPGRADE_FLAVOUR_SAP_HANA): '8.4',
+
+    # unsupported fallback paths for RHEL 8
+    ('8', LEAPP_UPGRADE_FLAVOUR_DEFAULT): '9.0',
+}
+
+
+def _get_major_version(version):
```

Comment:
Might be a useful helper function in libraries/config/version.py and can be used here
Ofc you can't use 'get_target_major_version' but this helper could be there, to keep the logic for version related functions in one place. 

Again, a follow up